shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h

```diff
@@ -287,8 +300,37 @@ class FlutterPlatformViewsController {
   // Pushes the view id of a visted platform view to the list of visied platform views.
   void PushVisitedPlatformView(int64_t view_id) { visited_platform_views_.push_back(view_id); }
 
+  // Visible for testing.
+  void CompositeWithParams(int64_t view_id, const EmbeddedViewParams& params);
+
+  const EmbeddedViewParams& GetCompositionParams(int64_t view_id) const {
+    return current_composition_params_.find(view_id)->second;
+  }
+
  private:
-  using LayersMap = std::map<int64_t, std::vector<std::shared_ptr<FlutterPlatformViewLayer>>>;
+  struct LayerData {
+    SkRect rect;
+    int64_t view_id;
+    int64_t overlay_id;
+    std::shared_ptr<FlutterPlatformViewLayer> layer;
+  };
+
+  using LayersMap = std::map<int64_t, LayerData>;
+
+  /// Update the buffers and mutate the platform views in CATransaction on the platform thread.
+  void PerformSubmit(const LayersMap& platform_view_layers,
+                     std::map<int64_t, EmbeddedViewParams>& current_composition_params,
```

Comment:
For that matter, the view params map could also probably be `std::unordered_map`... you know, while you're in here... :P

I realise this isn't core to your change, so feel free to dump in a followup.

Reply:
Done